### DISCUSSION TITLE
How to solve TypeScript errors with @turf/turf - "Could not find a declaration file for module"

### DISCUSSION THREAD

**Discussion by bjornhansen** [Turfjs/turf#2521] [Q&A] [answered]
As far as I understand, there are included TypeScript types with the library, but my IDE (Jetbrains) is throwing TS errors when I install the library. Can anyone help me understand what is wrong, and if it's on my side what should I do?

What I did:

1. Ran `npm install @turf/turf`
2. Import a function: `import centroid from "@turf/centroid";`
3. Use the function (works fine)

**The error is:**

  TS7016: Could not find a declaration file for module `@turf/centroid`.`[local path]/node_modules/@turf/turf/dist/es/index.js` implicitly has an `any` type.
  
  There are types at `[local path]/node_modules/@turf/turf/index.d.ts`, but this result could not be resolved when respecting `package.json` exports . The `@turf/turf` library may need to update its package.json or typings.

**Comment by twelch** [comment, marked as answer]
Sounds similar to some errors that have gone by in other issues and potentially fixed in the latest 7.x alpha version of turf on npm. Give it a try?

**Reply by bjornhansen** [reply]
Thanks, @twelch! I just tried the v7 alpha and the error is gone. Looking forward to full v7 release, thank you for your work on this!
